### ISSUE TITLE
RCS1196 does not respect operator precedence

### ISSUE THREAD

**Issue by IanKemp** [JosefPihrt/Roslynator#174]
```csharp
void Main()
{
    string foobar = "foobar";
    string bar = "bar";

    string aStringThatMayBeNull;

    aStringThatMayBeNull = null;

    // original
    Console.WriteLine(StringExtensions.TrimEnd(aStringThatMayBeNull ?? foobar, bar)); // output: "foo"
    // RCS1196
    Console.WriteLine(aStringThatMayBeNull ?? foobar.TrimEnd(bar)); // output: "foo"
    // correct
    Console.WriteLine((aStringThatMayBeNull ?? foobar).TrimEnd(bar)); // output: "foo"

    aStringThatMayBeNull = "not null bar";

    // original
    Console.WriteLine(StringExtensions.TrimEnd(aStringThatMayBeNull ?? foobar, bar)); // output: "not null"
    // RCS1196
    Console.WriteLine(aStringThatMayBeNull ?? foobar.TrimEnd(bar)); // output: "not null bar" (WRONG)
    // correct
    Console.WriteLine((aStringThatMayBeNull ?? foobar).TrimEnd(bar)); // output: "not null"
}

public static class StringExtensions
{
    public static string TrimEnd(this string @this, string end)
    {
        return @this.Substring(0, @this.Length - end.Length);
    }
}
```

RCS1196 is triggered by the invocation to `StringExtensions.TrimEnd`:

```csharp
StringExtensions.TrimEnd(aStringThatMayBeNull ?? foobar, bar)
```

and suggests to rewrite it as:

```csharp
aStringThatMayBeNull ?? foobar.TrimEnd(bar)
```

This is incorrect because it ignores operator precedence - RCS1196's suggestion is equivalent to writing the following:

```csharp
aStringThatMayBeNull ?? (foobar.TrimEnd(bar))
```

However, the original intent of the code (and the way RCS1196 **should** rewrite it) is:

```csharp
(aStringThatMayBeNull ?? foobar).TrimEnd(bar)
```
